File: src/test/java/io/icednut/algorithm/exercise/bit/leet1310/SolutionTest.java
package io.icednut.algorithm.exercise.bit.leet1310;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SolutionTest {

    @Test
    void test01() {
        var solution = new Solution();

        var output = solution.xorQueries2(
                new int[]{1, 3, 4, 8},
                new int[][]{
                        new int[]{0, 1},
                        new int[]{1, 2},
                        new int[]{0, 3},
                        new int[]{3, 3},
                }
        );

        assertArrayEquals(
                new int[]{2, 7, 14, 8},
                output
        );
    }

    @Test
    void test02() {
        var solution = new Solution();

        var output = solution.xorQueries2(
                new int[]{4, 8, 2, 10},
                new int[][]{
                        new int[]{2, 3},
                        new int[]{1, 3},
                        new int[]{0, 0},
                        new int[]{0, 3},
                }
        );

        assertArrayEquals(
                new int[]{8, 0, 4, 4},
                output
        );
    }

    @Test
    void test03() {
        var solution = new Solution();

        var output = solution.xorQueries2(
                new int[]{16},
                new int[][]{
                        new int[]{0, 0},
                        new int[]{0, 0},
                        new int[]{0, 0},
                        new int[]{0, 0},
                }
        );

        assertArrayEquals(
                new int[]{16, 16, 16, 16},
                output
        );
    }
}

File: src/test/kotlin/io/icednut/algorithm/exercise/etc/leet874/SolutionTest.kt
package io.icednut.algorithm.exercise.etc.leet874

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class Solution3Test {

    @Test
    fun test01() {
        val solution = Solution()

        val result = solution.robotSim(
            commands = intArrayOf(4, -1, 3),
            obstacles = arrayOf<IntArray>()
        )

        assertEquals(25, result)
    }

    @Test
    fun test02() {
        val solution = Solution()

        val result = solution.robotSim(
            commands = intArrayOf(4, -1, 4, -2, 4),
            obstacles = arrayOf(
                intArrayOf(2, 4)
            )
        )

        assertEquals(65, result)
    }

    @Test
    fun test03() {
        val solution = Solution()

        val result = solution.robotSim(
            commands = intArrayOf(6, -1, -1, 6),
            obstacles = arrayOf()
        )

        assertEquals(36, result)
    }
}